src/strings/shiftjis.c

```diff
@@ -350,7 +350,7 @@ MVMuint32 MVM_string_shiftjis_decodestream(MVMThreadContext *tc, MVMDecodeStream
         MVMuint8 *bytes = (MVMuint8 *)cur_bytes->bytes;
         while (pos < cur_bytes->length || repl_pos) {
             MVMGrapheme32 graph;
-            MVMCodepoint codepoint;
+            MVMCodepoint codepoint = 0;
```

Comment:
The compiler is right: codepoint can be uninitialized if decode_handler returns DECODE_ERROR and we are in replacement mode. In this case when we pass codepoint to MVM_string_decode_stream_maybe_sep it's either uninitialized memory or the previous iteration's value. Neither would be helpful.

The one thing I'm not sure about though is if 0 is a proper value to initialize with. After all 0 is a valid code point, i.e. that of the NULL character. That's why we have the graph_is_set flag to avoid the same ambiguity there. Actually, NULL is an entirely appropriate separator character in some situations. Think `xargs -0`

Reply:
What about moving the var up a level and initializing to NULL?

Reply:
@samcv ping

Reply:
Line 432 is currently:
```
if (MVM_string_decode_stream_maybe_sep(tc, seps, codepoint)) {
```
should be
```
if (handler_rtrn == DECODE_CODEPOINT && MVM_string_decode_stream_maybe_sep(tc, seps, codepoint)) {
```

Reply:
I'll try that.

Reply:
Please don't initialize `codepoint` to `0`. If you do that, then we won't get any compiler warnings that it is used before initialization. Because if `codepoint` is used before initialized, that indicates a programming error.

Reply:
I did still get the original warnings when I did NOT initialize it. I'll try again without initializing it...

Reply:
I used a gcc pragma to turn off the uninitialized warning. No warnings for our source now.

Reply:
However, now there are warnings about unknown pragmas in clang and msvc. Unfortunately you need to do something like https://github.com/MoarVM/MoarVM/blob/master/src/profiler/heapsnapshot.c#L916-L929 to be clean across all three compilers.

Reply:
Thanks, I'll apply the pragmas in like manner.

Reply:
I have tried several ways to separate pre-processing options for 'gcc' and 'clang' by macro defs '__GNUC__' and '__clang__' but I haven't been successful. It's a chicken or egg situation since I can fix it for either compiler but not the other. At this point I'm not sure what to try next.  

My gut tells me that something is awry with the C preprocessor or at least how or when it is called in the build process. Clang and GCC pragma defs are leaking into each other's pre-processing definitions.

Reply:
Some progress: appears both compilers define __GNUC__ so I have to find the real distinguishing macro between the two to continue.

Reply:
That's above my pay grade to determine whether the compiler is correct on the uninitialized warning, but at least the warnings can be squelched if desired.